dt_model/model/abstract_model.py

```diff
@@ -0,0 +1,26 @@
+from __future__ import annotations
+
+from dt_model.symbols.context_variable import ContextVariable
+from dt_model.symbols.presence_variable import PresenceVariable
+from dt_model.symbols.index import Index
+from dt_model.symbols.constraint import Constraint
+
+
+class AbstractModel:
+    def __init__(
+            self,
+            name,
+            cvs: list[ContextVariable],
+            pvs: list[PresenceVariable],
+            indexes: list[Index],
+            capacities: list[Index],
+            constraints: list[Constraint],
```

Comment:
Another future improvement here might be to switch to `Sequence` rather than using `list`, which would allow to more flexibly initialize those fields, unless we really need the (mutable) `list` semantics.